sceptre/config/reader.py

```diff
@@ -395,16 +414,43 @@ def _recursive_read(
 
         # Read config file and overwrite inherited properties
         child_config = self._render(directory_path, filename, config_group) or {}
+        child_config.update(self._get_merge_with_stratgies(config, child_config))
+        config.update(child_config)
+        return config
 
-        for config_key, strategy in CONFIG_MERGE_STRATEGIES.items():
-            value = strategy(config.get(config_key), child_config.get(config_key))
+    def _get_merge_with_stratgies(self, left: dict, right: dict) -> dict:
+        """
+        Returns a new dict with only the merge values of the two inputs, using the
+        merge strategies defined for each key.
+        """
+        merge = {}
+
+        # Then apply the merge strategies to each item
+        for config_key, default_strategy in CONFIG_MERGE_STRATEGIES.items():
+            strategy = default_strategy
+            override_key = f"{config_key}_inheritance"
+            if override_key in CONFIG_MERGE_STRATEGIES:
+                name = CONFIG_MERGE_STRATEGIES[override_key](
+                    left.get(override_key), right.get(override_key)
+                )
+                if not name:
+                    pass
+                elif name not in strategies.SELECTABLE:
+                    raise SceptreException(
+                        f"{name!r} is not a valid inheritance strategy"
```

Comment:
i have not seen `!r` before but the doc says that it's not needed in fstrings? https://peps.python.org/pep-0498/#s-r-and-a-are-redundant

Reply:
Hmm, I hadn't seen that section before - I read that as, `!r` is not strictly necessary because you can just do `f"{repr(name)}".

This is a personal habit of mine, the rational here is to clearly present what the incorrect variable value was because it will use the repr of the value.  So for example if the value was an empty string the error would be `"'' is not a valid inheritance strategy"` or if for some reason the value had trailing whitespace `"'dict_merge ' is not a valid inheritance strategy"`.

I'm happy to remove `!r` and just let it be the str of name if that's preferable?

Reply:
i think it's more common without thhe `!r` therefore probably easier to understand for most people so i would remove.